src/smt/theory_str_noodler/theory_str_noodler.cpp

```diff
@@ -968,12 +968,17 @@ namespace smt::noodler {
                 auto [noodler_lengths, precision] = dec_proc->get_lengths();
                 lengths = len_node_to_z3_formula(noodler_lengths);
                 lbool is_lengths_sat = check_len_sat(lengths);
-
-                // we assume that precision != LenNodePrecision::OVERAPPROX
                 
                 if (is_lengths_sat == l_true) {
                     STRACE("str", tout << "len sat " << mk_pp(lengths, m) << std::endl;);
                     last_run_was_sat = true;
+                    // save the current assignment to catch it during the loop protection
+                    block_curr_len(lengths, true, false);
```

Comment:
I would also remove this, only if you actually want to add this in this PR (this was replaced with the `last_run_was_sat = true` line).

Reply:
done